config/samples/lightgbm-dist/Dockerfile

```diff
@@ -11,14 +11,14 @@ RUN apt-get update && \
     gcc \
     g++ \
     git \
-    wget && \
+    curl && \
     # python environment
-    wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
-    /bin/bash Miniconda3-latest-Linux-x86_64.sh -f -b -p $CONDA_DIR && \
+    curl -sL https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -o conda.sh && \
+    /bin/bash conda.sh -f -b -p $CONDA_DIR && \
     export PATH="$CONDA_DIR/bin:$PATH" && \
     conda config --set always_yes yes --set changeps1 no && \
     # lightgbm
-    conda install -q -y numpy==1.19.1 scipy==1.5.2 scikit-learn==0.23.2 pandas==1.1.3 && \
+    conda install -q -y numpy scipy scikit-learn pandas && \
```

Comment:
why do you remove the version here? any version is ok?

Reply:
@merlintang As I can see current Package versions have these conflicts:
```
#6 78.91 Found conflicts! Looking for incompatible packages.
#6 78.91 This can take several minutes.  Press CTRL-C to abort.
#6 78.91 failed
#6 78.91 
#6 78.91 UnsatisfiableError: The following specifications were found
#6 78.91 to be incompatible with the existing python installation in your environment:
#6 78.91 
#6 78.91 Specifications:
#6 78.91 
#6 78.91   - numpy==1.19.1 -> python[version='>=3.6,<3.7.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']
#6 78.91   - pandas==1.1.3 -> python[version='>=2.7,<2.8.0a0|>=3.5,<3.6.0a0']
#6 78.91   - scipy==1.5.2 -> python[version='>=2.7,<2.8.0a0|>=3.5,<3.6.0a0']
#6 78.91 
#6 78.91 Your python: python=3.9
#6 78.91 
#6 78.91 If python is on the left-most side of the chain, that's the version you've asked for.
#6 78.91 When python appears to the right, that indicates that the thing on the left is somehow
#6 78.91 not available for the python version you are constrained to. Note that conda will not
#6 78.91 change your python version to a different minor version unless you explicitly specify
#6 78.91 that.
#6 78.91 
#6 78.91 The following specifications were found to be incompatible with each other:
#6 78.91 
#6 78.91 Output in format: Requested package -> Available versions
#6 78.91 
#6 78.91 Package setuptools conflicts for:
#6 78.91 scikit-learn==0.23.2 -> joblib[version='>=0.11'] -> setuptools
#6 78.91 python=3.9 -> pip -> setuptools
#6 78.91 
#6 78.91 Package numpy conflicts for:
#6 78.91 scikit-learn==0.23.2 -> scipy -> numpy[version='>=1.11.3,<2.0a0|>=1.15.1,<2.0a0|>=1.9.3,<2.0a0']
#6 78.91 scipy==1.5.2 -> numpy[version='>=1.14.6,<2.0a0|>=1.16.6,<2.0a0']
#6 78.91 scikit-learn==0.23.2 -> numpy[version='>=1.14.6,<2.0a0|>=1.16.6,<2.0a0']
#6 78.91 pandas==1.1.3 -> numpy[version='>=1.15.4,<2.0a0|>=1.16.6,<2.0a0']
#6 78.91 scipy==1.5.2 -> mkl-service[version='>=2,<3.0a0'] -> numpy[version='>=1.11.3,<2.0a0']
#6 78.91 
#6 78.91 Package six conflicts for:
#6 78.91 numpy==1.19.1 -> mkl-service[version='>=2,<3.0a0'] -> six
#6 78.91 pandas==1.1.3 -> python-dateutil[version='>=2.7.3'] -> six[version='>=1.5']
#6 78.91 scipy==1.5.2 -> mkl-service[version='>=2,<3.0a0'] -> six
#6 78.91 
#6 78.91 Package tzdata conflicts for:
#6 78.91 python=3.9 -> tzdata
#6 78.91 scikit-learn==0.23.2 -> python[version='>=3.9,<3.10.0a0'] -> tzdata
#6 78.91 scipy==1.5.2 -> python[version='>=3.9,<3.10.0a0'] -> tzdata
#6 78.91 pandas==1.1.3 -> python[version='>=3.9,<3.10.0a0'] -> tzdata
#6 78.91 
#6 78.91 Package intel-openmp conflicts for:
#6 78.91 scipy==1.5.2 -> mkl[version='>=2019.4,<2021.0a0'] -> intel-openmp
#6 78.91 numpy==1.19.1 -> mkl[version='>=2019.4,<2021.0a0'] -> intel-openmpThe following specifications were found to be incompatible with your system:
#6 78.91 
#6 78.91   - feature:/linux-64::__glibc==2.23=0
#6 78.91   - feature:|@/linux-64::__glibc==2.23=0
#6 78.91   - numpy==1.19.1 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
#6 78.91   - pandas==1.1.3 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
#6 78.91   - scikit-learn==0.23.2 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
#6 78.91   - scipy==1.5.2 -> libgcc-ng[version='>=7.3.0'] -> __glibc[version='>=2.17']
#6 78.91 
#6 78.91 Your installed version is: 2.23
#6 78.91 
#6 78.91 
```

Let me try to bump the latest available version and try to build the Docker image.

Reply:
I was able to build and run example with the latest versions: `numpy==1.20.3 scipy==1.6.2 scikit-learn==0.24.2 pandas==1.3.0`.
Does it sound good to you @merlintang ?

Reply:
ok. thanks for your contribution